#!/usr/bin/env python
# Follower.cfg
PACKAGE = "pmadmu_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("lookahead_distance", double_t, 0, "Higher distance -> smoother curves when driving but might leave path.", 1.2, 0.1, 10.0)
gen.add("lookahead_time", double_t, 0, "Higher value -> earlier replanning when a collision with a previously unknown obstacle is inbound.", 10.0, 1.0, 20.0)
gen.add("max_linear_speed", double_t, 0, "Max driving speed [m/s]", 1.0, 0.0, 3.0)
gen.add("max_angular_speed", double_t, 0, "Max rotation speed [rad/s]", 1.5, 0.0, 5.0)

gen.add("goal_tolerance", double_t, 0, "Distance at which the goal position is considered to be reached [m]", 0.05, 0.0, 0.5)
gen.add("rotation_tolerance", double_t, 0, "Angle at which the goal rotation is considered to be reached [rad]", 0.002, 0.0, 0.1)
gen.add("slowdown_angle", double_t, 0, "Angle at which the slowdown begins [rad]", 0.25, 0.0, 1.0)

gen.add("slowdown_x", double_t, 0, "Defines a box's x-axis that causes slowdowns if objects enter it [m]", 3.0, 0.0, 10.0)
gen.add("slowdown_y", double_t, 0, "Defines a box's y-axis that causes slowdowns if objects enter it [m]", 1.4, 0.0, 10.0)
gen.add("stopping_x", double_t, 0, "Defines a box's x-axis that causes a stop if objects enter it [m]", 1.50, 0.0, 10.0)
gen.add("stopping_y", double_t, 0, "Defines a box's y-axis that causes a stop if objects enter it [m]", 0.90, 0.0, 10.0)
gen.add("robot_size_x", double_t, 0, "Robot size along x-axis [m]", 1.25, 0.1, 5.0)
gen.add("robot_size_y", double_t, 0, "Robot size along y-axis [m]", 0.85, 0.1, 5.0)


exit(gen.generate(PACKAGE, "path_follower", "Follower"))